opentreemap/treemap/templatetags/form_extras.py

```diff
@@ -27,7 +27,7 @@
 # model.udf:field name, can't be done in the grammar as it can't be checked
 # until looked up in the context
 _identifier_regex = re.compile(
-    r"^[a-zA-Z_.\-]+(?:udf\:[\w '\._-]+|[a-zA-Z0-9_\-]+)$")
+    r"^[a-zA-Z_.\-]+(?:udf\:.+|[a-zA-Z0-9_\-]+)$")
```

Comment:
If synchronization is the only thing we are worried about, we can make a single source of truth

```
UDF_NAME_PATTERN = r'[^_%.]+'
_UDF_NAME_REGEX = re.compile(UDF_NAME_PATTERN)
```

```
_identifier_regex = re.compile(
    r"^[a-zA-Z_.\-]+(?:udf\:" + UDF_NAME_PATTERN + r"|[a-zA-Z0-9_\-]+)$")
```

If you don't want to do that, I would like to embed your concern in a comment above the "looser" regex.